filebeat/processors/annotate/kubernetes/indexing.go

```diff
@@ -0,0 +1,75 @@
+package kubernetes
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors/annotate/kubernetes"
+)
+
+func init() {
+	kubernetes.Indexing.AddMatcher("logs_path", newLogsPathMatcher)
+
+	//Add a container indexer by default.
+	cfg := common.NewConfig()
+	container, err := kubernetes.NewContainerIndexer(*cfg)
+	if err == nil {
+		kubernetes.Indexing.AddDefaultIndexer(container)
+	} else {
+		logp.Err("Unable to load indexer plugin due to error: %v", err)
+	}
+
+	//Add a log path matcher which can extract container ID from the "source" field.
+	logsPathMatcher, err := newLogsPathMatcher(*cfg)
+	if err == nil {
+		kubernetes.Indexing.AddDefaultMatcher(logsPathMatcher)
```

Comment:
Why not just adding the registered matcher/indexer name here, instead of registering an instance? like `AddDefaultIndexer("logs_path")`

This would force us to register it before marking it as default